Jdk11

physikerwelt opened the pull request.

physikerwelt committed:
Add test for highlighting MathML by content identifier hash

physikerwelt committed:
Add utility class for identifiers

physikerwelt committed:
Implement matching logic

physikerwelt committed:
Implement actual highlighting

physikerwelt committed:
Document code

physikerwelt committed:
Simplify testing

physikerwelt committed:
Add more tests

physikerwelt committed:
Address vstanges comments

physikerwelt committed:
Decomplexify algorithm

physikerwelt committed:
Decomplexify highlightFirstIdentifier

physikerwelt committed:
Use robust method to get identifiers

physikerwelt committed:
Minor: Clean up imports

physikerwelt committed:
Decomplexify highlightFirstIdentifier further

Signed-off-by: Moritz Schubotz (physikerwelt) <wiki@physikerwelt.de>

physikerwelt committed:
specify namespace for ci filter

physikerwelt committed:
Improve test coverage

Signed-off-by: Moritz Schubotz (physikerwelt) <wiki@physikerwelt.de>

physikerwelt committed:
Remove oracle from the build matrix

See travis-ci/travis-ci#10290

Signed-off-by: Moritz Schubotz (physikerwelt) <wiki@physikerwelt.de>

physikerwelt committed:
Update dependencies

physikerwelt committed:
WIP update xml dependencies

It is not clear how to get the DOMInputImpl or how to create the LSInput Object.

Any Ideas?

Signed-off-by: Moritz Schubotz (physikerwelt) <wiki@physikerwelt.de>

physikerwelt commented on mathml-core/src/main/java/com/formulasearchengine/mathmltools/mml/MathDoc.java:
```diff
@@ -163,10 +163,10 @@ public static String extractNamespacePrefix(StringBuffer rawXml) {
         SchemaInput schemaInput = new SchemaInput().invoke();
         InputSource inputSource = schemaInput.getInputSource();
-        final DOMInputImpl input = new DOMInputImpl();
+        final LSInput input = new DOMInputImpl();
```
Here I got stuck. No Idea how to get a new DOMInputImpl. @AndreG-P @vstange

AndreG-P replied:
Is this maybe something you are looking for:
[DOMImplementationLS](https://docs.oracle.com/en/java/javase/11/docs/api/java.xml/org/w3c/dom/ls/DOMImplementationLS.html)
``` java 
final LSInput input = DOMImplementationLS.createLSInput();
```

AndreG-P committed:
Upgrade maven to Java 11, fix errors with LSInput and in tests
Upgrade maven to Java 11 (including maven-compiler-plugin, maven surefire, and jacoco).
Further fixed the error with LSInput mentioned my @physikerwelt.
Further updated test cases because the expected results did not match the return values from current mathoid and latexml versions.

AndreG-P committed:
Fix potential null pointer in MathDoc
